docs/docs/guides/integrate/token-exchange.mdx

```diff
@@ -0,0 +1,894 @@
+  "token_type": "Bearer",
+  "expires_in": 43199,
+  "scope": "openid profile email",
+  "id_token": "eyJhbGciOiJSUzI1NiIsImtpZCI6IjI1OTM3OTQwMTIwNzA1NDMzOCIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjkwMDAiLCJzdWIiOiIyNTkyNDIwMzkzNzg0NDQyOTAiLCJhdWQiOlsiMjU5MjU0NDA5MzIwNTI5OTIyQHBvcnRhbCIsIjI1OTI5Nzc3MzUwODE2NTYzNEBwb3J0YWwiLCIyNTkyNTQzMTcwNzkzMzA4MTgiLCIyNTkyNTQwMjAzNTc0ODg2NDIiLCIyNTkyNTY1ODgxMjcxNzQ2NTgiLCIyNTc3ODY5OTEyNDcyOTQ0NjgiXSwiZXhwIjoxNzExMTQ1OTUxLCJpYXQiOjE3MTExMDI3NTEsImF6cCI6IjI1OTI5Nzc3MzUwODE2NTYzNEBwb3J0YWwiLCJjbGllbnRfaWQiOiIyNTkyOTc3NzM1MDgxNjU2MzRAcG9ydGFsIiwiYWN0Ijp7ImlzcyI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTAwMCIsInN1YiI6IjI1OTI0MTk0NDY1NDI4Mjc1NCJ9LCJhdF9oYXNoIjoiMENlN0pUMExHYUVJTmxwQVRIYzFRQSIsIm5hbWUiOiJlbmQgdXNlciIsImdpdmVuX25hbWUiOiJlbmQiLCJmYW1pbHlfbmFtZSI6InVzZXIiLCJuaWNrbmFtZSI6ImVuZC11c2VyIiwiZ2VuZGVyIjoiZmVtYWxlIiwibG9jYWxlIjoiZW4iLCJ1cGRhdGVkX2F0IjoxNzExMDE2Mjk2LCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJlbmQtdXNlciIsImVtYWlsIjoidGltK2VuZC11c2VyQHppdGFkZWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWV9.cMWoJBIPeakjtXWzsW3SGOAjMl27E0q8iePXtUHGueSUMhPibpOn7JiKd7VaZhgMDqN6c5TCU0EErdVm-6bc4SkxqrnYFjX4YIOygoTSbNzqkiOss6ZpcAGHt_RAd-i6NGcEm2_Fqp-EUO45V7jBEWgo3O4XLHsCVV1LQFpCHaSPK0ZtmjmNw-s-UKKF-kdSLLBpYKEUNmWGSMp3MqMgKLwl0SKFOiMY_HmBb-zSDRGN6s68b9Ays6Edxt-EnQ0pfR0TYFbnVSBQCqi5VXt3AcdnV1LRFQWi8ux6YTOiU10fZ3jbOiDjfS85bEKl9Nq5mhxVn9VsO4IiynjA9ZmlLQ"
+}
+```
+
+And again the access token claims:
+
+```json
+{
+  "iss": "http://localhost:9000",
+  "sub": "259242039378444290",
+  "aud": [
+    "259254409320529922@portal",
+    "259297773508165634@portal",
+    "259254317079330818",
+    "259254020357488642",
+    "259256588127174658",
+    "257786991247294468"
+  ],
+  "exp": 1711145951,
+  "iat": 1711102751,
+  "nbf": 1711102751,
+  "jti": "259387086463827970",
+  "act": {
+    "iss": "http://localhost:9000",
+    "sub": "259241944654282754"
+  }
+}
+```
+
+### Other usage examples
+
+Above we gave some of the most staightforward usecases. Of course, you can combine these examples to:
+
+- Impersonate and change the token type
+- Impersonate and change scope
+- Impersonate and reduce audience
+- Impersonate, change the token type, scope and audience
+
+## Audit trail
+
+In the user view of the console we can see whenever a new access token is created for a user.
+The existing `Access Token created` event is also used in the case of a token exchange.
+When there was an `actor_token` present during token exchange, we also log a `User impersonated` event.
+
+![Screenshot showing the user audit log with token creation and impersonation](/img/guides/token-exchange/user-audit-log.png)
+
+In the instance event list the `User impersonated` carries the actor in the payload:
+
+```json
+{
+  "actor": {
+    "issuer": "http://localhost:9000",
+    "user_id": "259241944654282754"
+  },
+  "applicationId": "259297773508165634@portal",
+  "authTime": "0001-01-01T00:00:00Z",
+  "expiration": "0001-01-01T00:00:00Z"
```

Comment:
i wondered why these are empty and looks like we use the wrong event type currently.
should be `UserImpersonatedEvent` instead of `UserTokenAddedEvent` in:
https://github.com/zitadel/zitadel/blob/30ff0dd1eddf44520ce89ead0e722ffbc1a6da49/internal/repository/user/user.go#L301-L316